chaincode/compute_plan.go

```diff
@@ -413,20 +413,23 @@ func UpdateComputePlanState(db *LedgerDB, ComputePlanID, tupleStatus, tupleKey s
 	if err != nil {
 		return err
 	}
-	statusUpdated := cp.UpdateStatus(tupleStatus)
+	stateUpdated := cp.UpdateState(tupleStatus)
 	doneModels := []string{}
-	if tupleStatus == StatusDone {
-		doneModels, err = cp.CheckDoneIntermediaryModel(db)
+	if tupleStatus == StatusDone && cp.CleanModels {
+		doneModels, err = cp.UpdateIntermediaryModelsInuse(db)
+		stateUpdated = true
 	}
 
 	if err != nil {
 		return err
 	}
-	if statusUpdated || len(doneModels) != 0 {
+	if stateUpdated || len(doneModels) != 0 {
 		err = db.AddComputePlanEvent(ComputePlanID, cp.State.Status, doneModels)
 		if err != nil {
 			return err
 		}
+	}
+	if stateUpdated {
```

Comment:
Why not saving the state in the case the intemediary model list has been updated? Because if `doneModels` is not empty, then the list of _models in use_ present in the compute plan state has also been updated (cf [here](https://github.com/SubstraFoundation/substra-chaincode/blob/89b489257e22c79dc1905136295e8cafac45484a/chaincode/compute_plan.go#L508)) and therefore this compute plan state need to be saved.

Reply:
I agree, let me push an update.